core/src/main/scala/io/snappydata/sql/catalog/impl/StoreHiveCatalog.scala

```diff
@@ -434,24 +434,40 @@ class StoreHiveCatalog extends ExternalCatalog with Logging {
       }
     }
 
+    private def maskPassword(s: String): String = {
+      SnappyExternalCatalog.PASSWORD_MATCH.replaceAllIn(s, "xxx")
+    }
+
+    // Mask access key and secret access key in case of S3 URI
+    private def maskLocationURI(locURI: String): String = {
+      val maskedSrcPath = if (locURI.toLowerCase().startsWith("s3a://") ||
+          locURI.toLowerCase().startsWith("s3://") ||
+          locURI.toLowerCase().startsWith("s3n://") ) {
+        locURI.replace(locURI.slice(locURI.indexOf("//") + 2,
+          locURI.indexOf("@")), "****:****")
+      } else locURI
```

Comment:
change to "else maskPassword(locURI)" since location URI can also contain passwords